python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/vxlan-interface.j2

```diff
@@ -69,21 +69,36 @@
 {%     elif vxlan_config.vxlan.multicast.headend_replication is arista.avd.defined(false) %}
 | Multicast headend-replication | Disabled |
 {%     endif %}
-{%     if vxlan_config.vxlan.vlans is arista.avd.defined %}
+{%     if vxlan_config.vxlan.vlans is arista.avd.defined or vxlan_config.vxlan.vlan_range is arista.avd.defined %}
 
 ##### VLAN to VNI, Flood List and Multicast Group Mappings
 
 | VLAN | VNI | Flood List | Multicast Group |
 | ---- | --- | ---------- | --------------- |
-{%         for vlan in vxlan_config.vxlan.vlans | arista.avd.natural_sort('id') %}
-{%             set vlan_vni = vlan.vni | arista.avd.default('-') %}
-{%             set multicast_group = vlan.multicast_group | arista.avd.default('-') %}
+{%         set all_vlans = {} %}
+{%         for vlan in vxlan_config.vxlan.vlans %}
+{%             do all_vlans.setdefault(vlan.id, {}) %}
+{%             if vlan.vni is arista.avd.defined %}
+{%                 do all_vlans[vlan.id].update({"vni": vlan.vni}) %}
+{%             endif %}
+{%             if vlan.multicast_group is arista.avd.defined %}
+{%                 do all_vlans[vlan.id].update({"multicast_group": vlan.multicast_group}) %}
+{%             endif %}
 {%             if vlan.flood_vteps is arista.avd.defined %}
-{%                 set flood_list = vlan.flood_vteps | join('<br/>')  %}
-{%             else %}
-{%                 set flood_list = '-' %}
+{%                 do all_vlans[vlan.id].update({'flood_list': vlan.flood_vteps | join('<br/>')}) %}
 {%             endif %}
-| {{ vlan.id }} | {{ vlan_vni }} | {{ flood_list }} | {{ multicast_group }} |
+{%         endfor %}
+{%         if vxlan_config.vxlan.vlan_range.vlans is arista.avd.defined and vxlan_config.vxlan.vlan_range.vnis is arista.avd.defined %}
+{%             set vlans = vxlan_config.vxlan.vlan_range.vlans | arista.avd.range_expand %}
+{%             for vlan in vlans %}
+{%                 do all_vlans.setdefault(vlan, {}) %}
+{%                 set vnis = vxlan_config.vxlan.vlan_range.vnis | arista.avd.range_expand %}
+{%                 set vni = vnis[vlans.index(vlan)] %}
+{%                 do all_vlans[vlan].update({"vni": vni}) %}
+{%             endfor %}
```

Comment:
What if we move line [95](https://github.com/aristanetworks/avd/pull/4749/files#diff-0dec211966590837b2ff97babbfebc2202245f010b0d52c5e7b54a5bf26a1ccfR95) above "for" loop line [93](https://github.com/aristanetworks/avd/pull/4749/files#diff-0dec211966590837b2ff97babbfebc2202245f010b0d52c5e7b54a5bf26a1ccfR93) that way we won't have to expand `vnis` range for each vlan.

Reply:
Done.